pinot-tools/src/main/java/org/apache/pinot/tools/anonymizer/PinotDataAndQueryAnonymizer.java

```diff
@@ -287,6 +290,20 @@ public void buildGlobalDictionaries() throws Exception {
     LOGGER.info("Finished building global dictionaries. Time taken: {}secs", _timeToBuildDictionaries.elapsed(TimeUnit.SECONDS));
   }
 
+  private void getSchemaFromFirstSegment(String segmentDirectory) throws Exception {
+    LOGGER.info("Reading metadata from segment: " + segmentDirectory);
+    File segmentIndexDir = new File(segmentDirectory);
+    SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(segmentIndexDir);
+    if (_pinotSchema == null) {
+      // only do this for first segment
+      _pinotSchema = segmentMetadata.getSchema();
+      anonymizeColumnNames(_pinotSchema);
+      _avroSchema = getAvroSchemaFromPinotSchema(_pinotSchema);
+      LOGGER.info("Pinot schema: " + _pinotSchema.toPrettyJsonString());
```

Comment:
It'd be good not just print the schemas for this particular edge case but for all the cases.

Reply:
Done